portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,8 +41,14 @@ public void init() {
      private String convertToJson(ArrayList<String> messages) {
         String json = "{";
          for(int i = 0 ; i< messages.size(); i++){
+            if(i == messages.size()-1){
```

Comment:
If I'm reading this correctly, this `if` statement is meant to avoid adding the comma for the last element, right? If so, you might simplify this with something like:

```
for(...) {
    json += ...
    json += ...
    if(i < messages.size() - 1) {
      json += ", ";
    }
```

(That might not be exactly what you need to do, but hopefully that demonstrates the idea.)

Reply:
Yes, that's what I'm trying to do. Thanks, Kevin